src/Ocelot/Configuration/Creator/SecurityOptionsCreator.cs

```diff
@@ -1,12 +1,42 @@
-﻿using Ocelot.Configuration.File;
+﻿using NetTools; // <PackageReference Include="IPAddressRange" Version="5.0.0" />
+using Ocelot.Configuration.File;
+
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Ocelot.Configuration.Creator
 {
     public class SecurityOptionsCreator : ISecurityOptionsCreator
     {
         public SecurityOptions Create(FileSecurityOptions securityOptions)
         {
-            return new SecurityOptions(securityOptions.IPAllowedList, securityOptions.IPBlockedList);
+            var ipAllowedList = new List<string>();
+            var ipBlockedList = new List<string>();
+
+            foreach (var allowed in securityOptions.IPAllowedList)
+            {
+                if (IPAddressRange.TryParse(allowed, out var allowedIpAddressRange))
+                {
+                    var allowedIps = allowedIpAddressRange.AsEnumerable().Select(x => x.ToString());
```

Comment:
asEnumerable is probably useless

Reply:
@RaynaldM, could you track the changes and resolve code review issues, and finally give an approval if it's possible, please?
I'm not a PubSub notifier! :stuck_out_tongue_closed_eyes:

Reply:
The IPAddressRange type needs to be enumerate to get the IP list

Reply:
The definition:
```csharp
public class IPAddressRange : IEnumerable<IPAddress>, IEnumerable, IReadOnlyDictionary<string, string>, IEnumerable<KeyValuePair<string, string>>, IReadOnlyCollection<KeyValuePair<string, string>>, IEquatable<IPAddressRange>
{ }
```
The problem is in `IEnumerable` interface implementation. which is explicit interface implementation, **not implicit**:
```csharp
        // Summary:
        //     Returns the input typed as IEnumerable<IPAddress>
        public IEnumerable<IPAddress> AsEnumerable()
        {
            return Operator;
        }

        // implicit IEnumerable<IPAddress> interface implementation
        public IEnumerator<IPAddress> GetEnumerator()
        {
            return Operator.GetEnumerator();
        }

        // explicit IEnumerable interface implementation !!!
        IEnumerator IEnumerable.GetEnumerator()
        {
            return GetEnumerator();
        }
```
That's why we have to convert explicitly instead of compile-time substitution.
So, we have to call explicit conversion to the interface first like:
- `allowedIpAddressRange.AsEnumerable().Select(x => x.ToString())`
- `((IEnumerable<IPAddress>)allowedIpAddressRange).Select(x => x.ToString())`
- `((IReadOnlyDictionary<string, string>)allowedIpAddressRange).Values`

@RaynaldM I believe, the first variant is the most shortest.

Reply:
Well... I have a good new! Source code analyzer probably is confused by explicit implementations for `IEnumerable<T>`, it cannot implicitly substitute `Select<IPAddress, string>()` for the object. 😄 
So, we have to specify Type params explicitly, and compiler analyzer won't be confused! 😺 
Final version:
```suggestion
                    var allowedIps = allowedIpAddressRange.Select<IPAddress, string>(x => x.ToString());
```
@Fabman08  Enjoy! 😉

Reply:
@raman-m done! Thank you!